dask_ml/wrappers.py

```diff
@@ -119,6 +119,29 @@ def __init__(self, estimator=None, scoring=None):
         self.estimator = estimator
         self.scoring = scoring
 
+    def _check_array(self, X):
+        """Validate an array for post-fit tasks.
+
+        Parameters
+        ----------
+        X : Union[Array, DataFrame]
+
+        Returns
+        -------
+        same type as 'X'
+
+        Notes
+        -----
+        The following checks are applied.
+
+        - Ensure that the array is blocked only along the samples.
+        """
+        if isinstance(X, da.Array):
+            if X.ndim == 2 and X.numblocks[1] > 1:
+                logger.debug("auto-rechunking 'X'")
+                X = X.rechunk({0: "auto", 1: -1})
```

Comment:
I think it's an open question on whether or not we want to include `{0: 'auto'}` here.  I can imagine situations where either including or not including it would be problematic.

Reply:
Just to be clear, my PR only auto-rechunks when `X.numblcoks[1] > 1`. In your mini-example if `X.shape == (40, 10)`, we won't rechunk in `_check_array`.